streams/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java

```diff
@@ -153,14 +153,27 @@ public ByteBuffer subscriptionUserData(final Set<String> topics) {
                                                                    topics,
                                                                    standbyTasks,
                                                                    REBALANCE_PROTOCOL);
-            return new FutureSubscriptionInfo(
-                usedSubscriptionMetadataVersion,
-                LATEST_SUPPORTED_VERSION + 1,
-                taskManager.processId(),
-                activeTasks,
-                standbyTasks,
-                userEndPoint())
-                .encode();
+
+
+            if (usedSubscriptionMetadataVersion <= LATEST_SUPPORTED_VERSION) {
```

Comment:
requiring the FutureSubscriptionInfo to delegate to the "current" SubscriptionInfo introduces a lot of complexity... Why not instead just use the "future" implementation only for "future" versions, and use the "current" SubscriptionInfo for "supported" versions?